adobe/forest.hpp

```diff
@@ -269,7 +270,7 @@ class reverse_fullorder_iterator
     explicit reverse_fullorder_iterator(I x) : base_m(--x), edge_m(forest_leading_edge - base_m.edge()) {}
     template <typename U>
     reverse_fullorder_iterator(const reverse_fullorder_iterator<U>& x)
-        : base_m(x.base()), edge_m(x.edge_m) {}
+        : base_m(--x.base()), edge_m(forest_leading_edge - base_m.edge()) {}
```

Comment:
Since `base()` returns `std::next(base_m)`, we have to do the same steps in this ctor as the one in line 270.